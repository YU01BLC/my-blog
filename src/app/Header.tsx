import React from 'react';
import Image from 'next/image';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Link from 'next/link';
import BlogImage from '../../public/assets/blogIcon.png';

export default function Header() {
  return (
    <AppBar position='relative'>
      <Toolbar sx={{ bgcolor: '#37474f', borderBottom: 'solid' }}>
        <Image src={BlogImage} alt='blog icon' style={{ width: '32px', height: '32px', marginRight: '1em' }} />
        <Box
          sx={{
            display: 'flex',
            position: 'relative',
            justifyContent: 'space-between',
            width: '100%',
          }}
        >
          <Typography variant='h6' noWrap>
            <Link href='/'>My Blog</Link>
          </Typography>
          <Typography variant='h6' color='inherit' noWrap>
            <Button variant='contained' color='warning'>
              <Link href='/articles'>記事を書く</Link>
            </Button>
          </Typography>
        </Box>
      </Toolbar>
    </AppBar>
  );
}
