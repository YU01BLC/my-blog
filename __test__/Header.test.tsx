import { render, screen } from "@testing-library/react";
import Header from "../src/app/Header";
import userEvent from '@testing-library/user-event'
import "@testing-library/jest-dom";

describe("Page(Header)コンポーネントSnapshot", () => {
  it("snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});

// Headerコンポーネント画面表示要素確認テスト
describe("Page(Header)コンポーネント表示確認", () => {
  it("テキスト「My Blog」が表示されること", () => {
    render(<Header />);
    const headElement = screen.getByText("My Blog");
    expect(headElement).toBeInTheDocument();
  });

   it("テキスト「記事を書く」が表示されること", () => {
    render(<Header />);
    const headElement = screen.getByText("記事を書く");
    expect(headElement).toBeInTheDocument();
  });
});

// Headerコンポーネントページ遷移テスト
describe("ページ遷移テスト", () => {
  // app routerでのページ遷移に関するテスト方法が現状不明。一旦スキップで対応する。
  xit("テキスト「記事を書く」押下時に「Articles」コンポーネント要素のテキスト「articles」が表示されること", async () => {
    render(<Header />);
    userEvent.click(screen.getByText("記事を書く"))
    expect(await screen.getByText('articles')).toBeInTheDocument()
  });
});