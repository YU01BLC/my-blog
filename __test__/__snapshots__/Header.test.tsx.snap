// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page(Header)コンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-1vazdj7-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1o68ijm-MuiToolbar-root"
    >
      <img
        alt="blog icon"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; width: 32px; height: 32px; margin-right: 1em;"
        width="40"
      />
      <div
        class="MuiBox-root css-g6d0rx"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
        >
          <a
            href="/"
          >
            My Blog
          </a>
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-2l1qg7-MuiTypography-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium css-1qtu9q6-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/articles"
            >
              記事を書く
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </h6>
      </div>
    </div>
  </header>
</DocumentFragment>
`;

exports[`PageコンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionRelative css-1vazdj7-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1o68ijm-MuiToolbar-root"
    >
      <img
        alt="blog icon"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent; width: 32px; height: 32px; margin-right: 1em;"
        width="40"
      />
      <div
        class="MuiBox-root css-g6d0rx"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
        >
          <a
            href="/"
          >
            My Blog
          </a>
        </h6>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-2l1qg7-MuiTypography-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeMedium MuiButton-containedSizeMedium css-1qtu9q6-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <a
              href="/articles"
            >
              記事を書く
            </a>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </h6>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
